lib/ar2dto/has_dto.rb

```diff
@@ -22,17 +22,52 @@ def has_dto
       end
 
       # @api public
-      def to_dto
-        all.map(&:to_dto)
+      def to_dto(options = {})
+        all.map { |record| record.to_dto(options) }
       end
     end
 
     # Wrap the following methods in a module so we can include them only in the
     # ActiveRecord models that declare `has_dto`.
     module InstanceMethods
       # @api public
-      def to_dto
-        "#{self.class.name}DTO".constantize.new(attributes)
+      def to_dto(options = {})
+        options = apply_global_configs(options)
+        hash = serializable_hash(options&.except(:include))
+        hash = add_associations(hash, options)
+
+        "#{self.class.name}DTO".constantize.new(hash)
+      end
+
+      private
```

Comment:
I'd like to move all these private methods to a different class so they do not exist within an ActiveRecord object..

Reply:
Sounds fair 👍